contracts/bridge/BonusChef.sol

```diff
@@ -0,0 +1,640 @@
+        RewardPool storage pool = rewardPools[_rewardToken];
+        return pool.rewardRate;
+    }
+
+    /**
+        @notice Get total duration of a bonus reward pool
+        @param _rewardToken bonus reward token to check
+     */
+    function rewardsDuration(address _rewardToken)
+        external
+        view
+        returns (uint256)
+    {
+        RewardPool storage pool = rewardPools[_rewardToken];
+        return pool.rewardsDuration;
+    }
+
+    /**
+        @notice Get timestamp for the last payout in the bonus reward pool
+        @param _rewardToken bonus reward token to check
+     */
+    function lastUpdateTime(address _rewardToken)
+        external
+        view
+        returns (uint256)
+    {
+        RewardPool storage pool = rewardPools[_rewardToken];
+        return pool.lastUpdateTime;
+    }
+
+    /**
+        @notice Get total amount of bonus rewards per 1 LP token
+        in the MiniChef from the start of bonus pool until last update
+        @param _rewardToken bonus reward token to check
+     */
+    function rewardPerTokenStored(address _rewardToken)
+        external
+        view
+        returns (uint256)
+    {
+        RewardPool storage pool = rewardPools[_rewardToken];
+        return pool.rewardPerTokenStored;
+    }
+
+    /**
+        @notice Get amount of bonus rewards paid to user per 1 LP token
+        @param _rewardToken bonus reward token to check
+        @param _account user address
+     */
+    function userRewardPerTokenPaid(address _rewardToken, address _account)
+        external
+        view
+        returns (uint256)
+    {
+        RewardPool storage pool = rewardPools[_rewardToken];
+        return pool.userRewardPerTokenPaid[_account];
+    }
+
+    /**
+        @notice Get last stored amount of user's unpaid bonus rewards
+        @param _rewardToken bonus reward token to check
+        @param _account user address
+     */
+    function rewards(address _rewardToken, address _account)
+        external
+        view
+        returns (uint256)
+    {
+        RewardPool storage pool = rewardPools[_rewardToken];
+        return pool.rewards[_account];
+    }
+
+    /**
+        @notice Get all pending bonus rewards for user
+        @param _account user address
+     */
+    function pendingTokens(
+        uint256,
+        address _account,
+        uint256
+    ) external view override returns (IERC20[] memory, uint256[] memory) {
+        uint256 _activePoolsAmount = activeRewardPools.length;
+        IERC20[] memory _rewardTokens = new IERC20[](_activePoolsAmount);
+        uint256[] memory _rewardAmounts = new uint256[](_activePoolsAmount);
+        for (uint8 i = 0; i < _activePoolsAmount; i++) {
+            address _rewardToken = activeRewardPools[i];
+            _rewardTokens[i] = IERC20(_rewardToken);
+            _rewardAmounts[i] = _earned(
+                _rewardToken,
+                _account,
+                balanceOf(_account)
```

Comment:
Put this outside the for loop; if it's a constant value, no reason to constantly re-call it (and burn gas)

Reply:
I believe this is done because `address _rewardToken = activeRewardPools[i];` changes the earned amount. @ChiTimesChi can confirm

Reply:
Oh I did not see that, I'm dumb. Ignore me being dumb.

Reply:
> Put this outside the for loop; if it's a constant value, no reason to constantly re-call it (and burn gas)

Good catch! `balanceOf(account)` is indeed constant in the loop. No need to burn gas even in external view methods, I agree. Fixed in [3559dbd](https://github.com/synapsecns/synapse-contracts/pull/77/commits/3559dbd7d79cf5db5606bebd77bd5d66b1cd35c9)